app/views/banners/_form.html.erb

```diff
@@ -21,9 +21,10 @@
       <%= form.label :name %>
       <%= form.text_field :name, required: true %>
     </div>
-    <div class="form-check checkbox-style mb-20">
-      <%= form.check_box :active, class: 'form-check-input' %>
+    <div data-controller="checkbox reveal" data-action="<%= @has_alternate_active_banner && 'checkbox:toggle->reveal#toggle' %>" data-reveal-hidden-class="d-none" class="form-check checkbox-style mb-20">
+      <%= form.check_box :active, data: { action: "click->checkbox#toggle", checkbox_target: "checkbox" }, class: 'form-check-input' %>
       <%= form.label :active, "Active?", class: 'form-check-label' %>
+      <span data-reveal-target="item" class="text-danger<%= form.object.active && @has_alternate_active_banner ? '' : ' d-none' %>">Warning: This will replace your current active banner</span>
```

Comment:
Could you elaborate more on what the reveal controller does? My thought is that an alternative is to add a `hidden` css class to determine if the error message is shown or hidden. Does the reveal controller do more than that?

Reply:
As far as I know, it does not anything outside of hide/show/toggle. My initial impression was to do this manually, but I gathered `stimulus-components` was a pretty standard add.